lib/auth-middleware/auth_middleware/__init__.py

```diff
@@ -85,71 +95,93 @@ def authenticate(group_name=None, internal_token_only=False):
+        raise OAuthProblem(
+            'Token data not available for authorization. Most likely an authentication error.')
     else:
         id_info = g.user_token_data
 
         if group_name is not None:
-            # check if the group_name is list or string
-            if isinstance(group_name, str):
-                # make group name as list
-                group_name = [group_name]
-
-            is_authorize = False
-            for name in group_name:
-                if uiucedu_is_member_of in id_info:
-                    is_member_of = id_info[uiucedu_is_member_of]
-                    if name in is_member_of:
-                        is_authorize = True
-                        break
-
-            if is_authorize is False:
-                logger.warning("User not authorized.")
+            AUTH_ISSUER = os.getenv(
+                'AUTH_ISSUER', '').strip()
+            # So we are to check is a group membership is required.
+            # Get the membership check keys based on issuer
+            if id_info['iss'] == AUTH_ISSUER:
+                is_member_of_key = is_member_of_claim
+            else:
+                is_member_of_key = uiucedu_is_member_of
+
+            if is_member_of_key in id_info:
+                is_member_of = id_info[is_member_of_key]
+                print("groups: " + str(is_member_of))
+                if is_member_of_key == is_member_of_claim and group_name not in is_member_of:
+                    logger.warning(
+                        "user is not a member of the group " + group_name)
+                    raise OAuthProblem('Invalid token')
+            else:
+                logger.warning(is_member_of_key +
+                               " field is not present in the ID Token")
                 raise OAuthProblem('Invalid token')
 
 # Checks that the request has the right secret for this. This call is used initially and assumes that
 # the header contains the x-api-key. This (trivially) returns true of the verification worked and
 # otherwise will return various other exit codes.
+
+
 def verify_secret(request):
     key = request.headers.get(rokwire_api_key_header)
     if not key:
-        logger.warning("Request missing the " + rokwire_api_key_header + " header")
-        raise OAuthProblem('Missing API Key')  # missing header means bad request
+        logger.warning("Request missing the " +
+                       rokwire_api_key_header + " header")
+        # missing header means bad request
+        raise OAuthProblem('Missing API Key')
     # Assumption is that the key is a comma separated list of uuid's
     # This simply turns it in to a list and iterates. If the supplied key is in this list, true is returned
     # Otherwise, an error is raised.
-    keys = os.getenv('ROKWIRE_API_KEY').strip().split(',')
+    # keys = os.getenv('ROKWIRE_API_KEY').strip().split(',')
+    keys = os.getenv('ROKWIRE_API_KEY').strip()
```

Comment:
```suggestion
    keys = os.getenv('ROKWIRE_API_KEY').strip().split(',')
```
ROKWIRE_API_KEY contains a comma separated list of keys. The previous line that was commented out needs to be restored/uncommented.